go/vt/vttablet/onlineddl/executor.go

```diff
@@ -167,16 +169,32 @@ func (e *Executor) execQuery(ctx context.Context, query string) (result *sqltype
 }
 
 func (e *Executor) initSchema(ctx context.Context) error {
+	e.initMutex.Lock()
+	defer e.initMutex.Unlock()
+
+	if e.schemaInitialized {
```

Comment:
We are not allowed to assume that we've "done" something even if we did it. There is one use case where users restore an older snapshot of a mysql instance underneath us. This was the reason why we have to use withddl lazily. We can only know at the time of applying a DML.

Reply:
> There is one use case where users restore an older snapshot of a mysql instance underneath us. 

That's a good point.

Consider looking at how I solved this with `orchestrator`: it _always_ knows what schema to deploy. See `versionIsDeployed` function in `db.go`, which writes down the git `sha1` to the backend database.

Reply:
:+1: